src/mito2/src/read/scan_region.rs

```diff
@@ -0,0 +1,187 @@
+/// SeqScan o-- ProjectionMapper
+/// SeqScan -- SendableRecordBatchStream
+/// ```
+pub(crate) struct ScanRegion {
+    /// Version of the region at scan.
+    version: VersionRef,
+    /// Directory of SST files.
+    file_dir: String,
+    /// Object store that stores SST files.
+    object_store: ObjectStore,
+    /// Scan request.
+    request: ScanRequest,
+}
+
+impl ScanRegion {
+    /// Creates a [ScanRegion].
+    pub(crate) fn new(
+        version: VersionRef,
+        file_dir: String,
+        object_store: ObjectStore,
+        request: ScanRequest,
+    ) -> ScanRegion {
+        ScanRegion {
+            version,
+            file_dir,
+            object_store,
+            request,
+        }
+    }
+
+    /// Returns a [Scanner] to scan the region.
+    pub(crate) fn scanner(self) -> Result<Scanner> {
+        self.seq_scan().map(Scanner::Seq)
+    }
+
+    /// Scan sequentailly.
+    pub(crate) fn seq_scan(self) -> Result<SeqScan> {
+        let time_range = self.build_time_range_predicate();
+
+        let ssts = &self.version.ssts;
+        let mut total_ssts = 0;
+        let mut files = Vec::new();
+        for level in ssts.levels() {
+            total_ssts += level.files.len();
+
+            for file in level.files.values() {
+                // Finds SST files in range.
+                if file_in_range(file, &time_range) {
+                    files.push(file.clone());
+                }
+            }
+        }
+
+        let memtables = self.version.memtables.list_memtables();
+
+        debug!(
+            "Seq scan region {}, memtables: {}, ssts_to_read: {}, total_ssts: {}",
+            self.version.metadata.region_id,
+            memtables.len(),
+            files.len(),
+            total_ssts
+        );
+
+        let predicate = Predicate::try_new(
+            self.request.filters.clone(),
+            self.version.metadata.schema.clone(),
+        )
+        .context(BuildPredicateSnafu)?;
+        let mapper = match &self.request.projection {
+            Some(p) => ProjectionMapper::new(&self.version.metadata, p.iter().copied())?,
+            None => ProjectionMapper::all(&self.version.metadata)?,
+        };
+
+        let seq_scan = SeqScan::new(self.file_dir, self.object_store, mapper, self.request)
+            .with_time_range(Some(time_range))
+            .with_predicate(Some(predicate))
+            .with_memtables(memtables)
+            .with_files(files);
+
+        Ok(seq_scan)
+    }
+
+    /// Build time range predicate from filters.
+    fn build_time_range_predicate(&self) -> TimestampRange {
+        let time_index = self.version.metadata.time_index_column();
+        let unit = time_index
+            .column_schema
+            .data_type
+            .as_timestamp()
+            .expect("Time index must have timestamp-compatible type")
+            .unit();
+        TimeRangePredicateBuilder::new(&time_index.column_schema.name, unit, &self.request.filters)
```

Comment:
`self.request.filters` contains a unit (the literal expr), is it necessary to pass an extra `unit` here? cc @v0y4g3r

Reply:
> `self.request.filters` contains a unit (the literal expr), is it necessary to pass an extra `unit` here? cc @v0y4g3r

Unit of timestamp column is also required since we may find time unit in filters differs from timestamp column unit.